application/console/migrations/m181128_213015_add_do_not_disclose_column_to_user_table.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+use yii\db\Migration;
+
+/**
+ * Handles adding do_not_disclose to table `user`.
+ */
+class m181128_213015_add_do_not_disclose_column_to_user_table extends Migration
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function safeUp()
+    {
+        $this->addColumn('user', 'do_not_disclose', 'boolean not null default false');
```

Comment:
It seems like we have some data duplication between idp-pw-api and idp-id-broker. Why do they both need a `do_not_disclose` field?

Reply:
It was a design decision influenced by the way idp-pw-api keeps a copy of user data. I'm open to alternate ideas if you would like to discuss.

Reply:
I'll go ahead and look at that PR (and you may have addressed this already), but we'll probably need to make sure that either (1) pw-api gets the latest value for `do_not_disturb` from id-broker every time it uses the value or (2) id-broker sends `do_not_disturb` changes to pw-api every time it changes there in id-broker. I believe we will eventually allow this field to be set via id-sync, based on data from the HR system, so it could change it id-broker by a means other than pw-api.

Reply:
I'd like to discuss this with @fillup to make sure I'm not mixing things up here (re: changes to `do_not_disturb` in id-broker that would leave it out-of-sync with pw-api).

Reply:
Okay, @Schparky. @fillup talked me through some of the design decisions and trade-offs, and it's fine for you to go proceed with both pw-api and id-broker having a `do_not_disclose` field. The pw-api's copy seems to work effectively as a cache, since it's updated from id-broker on login. It also looks like id-broker's value will very likely only be updated through pw-api... the integrations with HR systems to add another field and train people how to use it added enough complexity that it's better for us to just keep that setting entirely within this IdP-in-a-Box system.

Reply:
Good. Thanks for taking a good amount of time to work through it. Much appreciated.